src/vs/workbench/services/textfile/browser/textFileService.ts

```diff
@@ -108,27 +110,29 @@ export abstract class TextFileService implements ITextFileService {
 		this.toUnbind.push(this.editorGroupService.onEditorsChanged(() => this.onEditorFocusChanged()));
 	}
 
-	private beforeShutdown(): boolean | TPromise<boolean> {
+	private beforeShutdown(quitRequested: boolean): boolean | TPromise<boolean> {
```

Comment:
@Tyriar can you create an issue on this so that we can further think about. You can put my name as well.